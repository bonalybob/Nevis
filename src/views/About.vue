<template>
  <base-layout>
    <div slot="menu">
      <a class="back" @click="$router.go(-1)">Back</a>
    </div>
    <div slot="main" class="main">
      <div class="card">
        <h2>System</h2>
        <p>Nevis Version: 4.0.0-beta</p>
        <p>System: {{ system }}</p>
        <p>Node Version: {{ node }}</p>
        <p>Chrome Version: {{ chromium }}</p>
        <p>Electron Version: {{ electron }}</p>
      </div>
      <div class="card">
        <h2>License</h2>
        <h3>MIT License</h3>
        <p>Copyright (c) 2018 Ben Brown</p>
        <p>
          Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation
          files (the" Software"), to deal in the Software without restriction, including without limitation the
          rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
          and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
          The above copyright notice and this permission notice shall be included in all copies or substantial
          portions of the Software.
        </p>
        <p>
          THE SOFTWARE IS PROVIDED"AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
          OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
          OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
          TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
          IN THE SOFTWARE.
        </p>
      </div>
    </div>
  </base-layout>
</template>

<script>
import BaseLayout from '@/components/BaseLayout'

export default {
  components: {
    'base-layout': BaseLayout,
  },

  data: () => ({
    electron: window.process.versions.electron,
    chromium: window.process.versions.chrome,
    node: window.process.versions.node,
  }),

  computed: {
    system: () => window.process.platform + ' ' + window.process.arch,
  },
}
</script>

<style lang="stylus" scoped>
.main
  p, h1, h2, h3, h4, h5, h6, a
    user-select: text
</style>
