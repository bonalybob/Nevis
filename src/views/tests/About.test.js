import { shallowMount } from '@vue/test-utils'
import About from '@/views/About'

test('Is a Vue Instance', () => {
  const wrapper = shallowMount(About, {
    stubs: ['router-link'],
    mocks: {
      $electron: { remote: { app: { getVersion: jest.fn(() => '5') } } },
    },
  })
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = shallowMount(About, {
    stubs: ['router-link'],
    mocks: {
      $electron: { remote: { app: { getVersion: jest.fn(() => '5') } } },
    },
  })
  expect(wrapper.element).toMatchSnapshot()
})
