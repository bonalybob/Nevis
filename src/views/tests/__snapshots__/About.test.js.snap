// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<main>
  <h1
    class="mx-10 mb-1"
  >
    <back-arrow-stub
      to=""
    />
    About
  
  </h1>
   
  <div
    class="shadow mx-12 mb-3 px-3 pt-2 pb-1"
  >
    <h2
      class="pb-1"
    >
      System
    </h2>
     
    <p>
      Nevis Version: 5
    </p>
     
    <p>
      System: win32 x64
    </p>
     
    <p>
      Node Version: 10.15.3
    </p>
     
    <p>
      Chrome Version: 
    </p>
     
    <p>
      Electron Version: 
    </p>
  </div>
   
  <div
    class="shadow mx-12 mb-3 px-3 pt-2 pb-1"
  >
    <h2
      class="pb-1"
    >
      License
    </h2>
     
    <h3
      class="pb-1"
    >
      MIT License
    </h3>
     
    <p>
      Copyright (c) 2019 Ben Brown
    </p>
     
    <p
      class="pb-1"
    >
      
      Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation
      files (the" Software"), to deal in the Software without restriction, including without limitation the
      rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
      and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
      The above copyright notice and this permission notice shall be included in all copies or substantial
      portions of the Software.
    
    </p>
     
    <p
      class="pb-1"
    >
      
      THE SOFTWARE IS PROVIDED"AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
      OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
      OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
      TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
      IN THE SOFTWARE.
    
    </p>
  </div>
</main>
`;
